engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintValidatorContextImpl.java

```diff
@@ -189,9 +193,14 @@ public ConstraintValidatorContext addConstraintViolation() {
 			if ( constraintViolationCreationContexts == null ) {
 				constraintViolationCreationContexts = CollectionHelper.newArrayList( 3 );
 			}
+			if ( !(expressionVariables == null || expressionVariables.isEmpty()) && !expressionLanguageEnabled ) {
+				LOG.expressionVariablesDefinedWithExpressionLanguageNotEnabled(
```

Comment:
Maybe add a test to check that this warning actually occurs?

Reply:
Yes, I need to add a bunch of tests.

Reply:
Done.